bonfire/bonfire.py

```diff
@@ -1016,6 +1073,198 @@ def _cmd_apps_what_depends_on(
     print("\n".join(found) or f"no apps depending on {component} found")
 
 
+@reservation.command("create")
+@click.option(
+    '--bot',
+    '-b',
+    is_flag=True,
+    help="Use this flag to skip the duplicate reservation check (for automation)",
+)
+@options(_reservation_process_options)
+@options(_timeout_option)
+def _create_new_reservation(bot, name, requester, duration, timeout):
+    def _err_handler(err):
+        msg = f"reservation failed: {str(err)}"
+        _error(msg)
+
+    try:
+        res = get_reservation(name)
+        # Name should be unique on reservation creation.
+        if res:
+            raise FatalError(
+                f"Reservation with name {name} already exists"
+            )
+
+        res_config = process_reservation(name, requester, duration)
+
+        log.debug("processed reservation:\n%s", res_config)
+
+        if not bot:
+            if check_for_existing_reservation(res_config["items"][0]["spec"]["requester"]):
+                _warn_of_existing(res_config["items"][0]["spec"]["requester"])
+
+        try:
+            res_name = res_config["items"][0]["metadata"]["name"]
+        except (KeyError, IndexError):
+            raise Exception(
+                "error parsing name of Reservation from processed template, "
+                "check Reservation template"
+            )
+
+        apply_config(None, list_resource=res_config)
+
+        ns_name = wait_on_reservation(res_name, timeout)
+    except KeyboardInterrupt as err:
+        log.error("aborted by keyboard interrupt!")
+        _err_handler(err)
+    except TimedOutError as err:
+        log.error("hit timeout error: %s", err)
+        _err_handler(err)
+    except FatalError as err:
+        log.error("hit fatal error: %s", err)
+        _err_handler(err)
+    except Exception as err:
+        log.exception("hit unexpected error!")
+        _err_handler(err)
+    else:
+        log.info(
+            "namespace '%s' is reserved by '%s' for '%s'",
+            ns_name,
+            res_config["items"][0]["spec"]["requester"],
+            duration,
+        )
+        click.echo(ns_name)
+
+
+@reservation.command("extend")
+@click.option(
+    '--namespace',
+    '-ns',
```

Comment:
Can we use '-n' for the short arg here since that is used for --namespace in all other commands?

Reply:
:+1: